### ISSUE TITLE
--ignore-pattern from CLI is ignored if it is set in eslintrc

### ISSUE THREAD

**Issue by Floriferous** [eslint/eslint#13236]
I'm using the eslint CLI in parallel with a `eslintrc.js` config. In the config I have a set of `ignorePatterns []`. We use symlinked directories in our repo, but we let eslint format each of the symlinks so it doesn't matter where you access that symlink from.

However, when running CLI commands, to speed things up, we'd like to ignore those symlinks. This is why I add a `--ignore-pattern` flag to the `eslint` command. However, when `ignorePatterns` is set in the config, and you specify a `--ignore-pattern` as a flag, it is ignored.

I expect the additional flag to be merged into the `ignorePatterns` array.

Here's a simple reproduction: https://github.com/Floriferous/eslint-ignore-issue

* **ESLint Version: 6.8.0**
* **Node Version: 12.14.0**
* **npm Version: 6.13.4**

**Comment by anikethsaha** [comment]
I think it would be better to use `.eslintignore` for your case as `--ignore-pattern` 's pattern adds up with the entries in `.eslintignore`'s files. 

[docs](https://eslint.org/docs/user-guide/command-line-interface#ignore-pattern)

**Comment by Floriferous** [comment]
Well then maybe we should unify the behavior of `eslintrc` and `eslintignore`. I specifically stopped using `.eslintignore` because it appeared to be exactly the same. 

Are there other such gotchas where they differ? Maybe it'd be worth documenting that.
